src/Helper/Service/AssetFactory.php

```diff
@@ -22,24 +26,23 @@ class AssetFactory implements FactoryInterface
      */
     public function __invoke(ContainerInterface $container, $name, ?array $options = null)
     {
-        $helper = new Asset();
-
+        /** @psalm-var mixed $config */
         $config = $container->get('config');
-        if (isset($config['view_helper_config']['asset'])) {
-            $configHelper = $config['view_helper_config']['asset'];
-            if (isset($configHelper['resource_map']) && is_array($configHelper['resource_map'])) {
-                $helper->setResourceMap($configHelper['resource_map']);
-            } else {
-                throw new Exception\RuntimeException('Invalid resource map configuration.');
-            }
-        }
+        $config = is_array($config) ? $config : [];
+
+        $helperConfig = $this->assertArray('view_helper_config', $config);
+        $helperConfig = $this->assertArray('asset', $helperConfig);
+        /** @psalm-var array<non-empty-string, non-empty-string> $resourceMap */
+        $resourceMap = $this->assertArray('resource_map', $helperConfig);
 
-        return $helper;
+        return new Asset($resourceMap);
     }
 
     /**
      * Create service
      *
+     * @deprecated
```

Comment:
We should explain the deprecation, in line with other deprecation comment requirements that @froschdesign made clear :)